vtex-apps/challenge-tp-condition
# Add implementation

#### What does this PR do? \*

Add implementation of a Challenge that checks if the user has the condition rules specified in the Trade Policy.

Important for B2B scenarios, where a Trade Policy is closed.

#### How to test it? \*

1. Open https://breno3--storecomponents.myvtex.com/?sc=3
2. You will be redirected to `/foobar`
3. Login using the user `breno@mailinator.com` `Abcd1234`
4. Go to the home
5. Now, you will not be redirected

Why is that, the Trade Policy 3 have the following setting:

![image](https://user-images.githubusercontent.com/284515/68232169-4447fd00-ffdb-11e9-98ab-97cb477b6334.png)

This tells the backend to check if the logged in user has the propery `approved` as `true` in its MasterData data.

#### Describe alternatives you've considered, if any. \*



#### Related to / Depends on \*

> **Depends on https://github.com/vtex-apps/store-graphql/pull/423**
